src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1

```diff
@@ -3703,6 +3708,53 @@ class Organization: ADOSVTBase
             $controlResult.LogException($_)
         }
         return $controlResult;
+    }
+
+    hidden [ControlResult] CheckCreatePermissionsForFeedAutomatedFix([ControlResult] $controlResult) {
+
+        try{            
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject   
+            # roleid : 1 (remove broader group permisson to Create Feed) , roleid : 2 (add broader group permisson to Create Feed)                    
+            $body = "["
+
+            if (-not $this.UndoFix)
+            {                                 
+                    if ($body.length -gt 1) {$body += ","}
+                    $body += @"
+                        {                            
+                            "identityId": "$($RawDataObjForControlFix.Id)",
+                            "role": 1,
+                            "identityDescriptor": "$($($RawDataObjForControlFix.Descriptor).Replace('\','\\'))"                        
+                        }
+"@;                                
+            }
+            else {                                 
+                    if ($body.length -gt 1) {$body += ","}
+                    $body += @"
+                        {                            
+                            "identityId": "$($RawDataObjForControlFix.Id)",
```

Comment:
Remove duplicate code only Role is changing, other code can be come above if else and role can be assign inside if else.